src/cloudflare/internal/images-api.ts

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) 2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+type Fetcher = {
+  fetch: typeof fetch;
+};
+
+type RawInfoResponse =
+  | { format: "image/svg+xml" }
+  | {
+      format: string;
+      file_size: number;
+      width: number;
+      height: number;
+    };
+export class TransformationResultImpl implements TransformationResult {
+  constructor(private readonly bindingsResponse: Response) {}
+
+  contentType(): string {
+    const contentType = this.bindingsResponse.headers.get("content-type");
+    if (!contentType) {
+      throw new Error("Content-type not set on binding response");
+    }
+
+    return contentType;
+  }
+
+  image(): ReadableStream<Uint8Array> {
+    return this.bindingsResponse.body || new ReadableStream();
+  }
+
+  public response(): Response {
+    return new Response(this.image(), {
+      headers: {
+        "content-type": this.contentType(),
+      },
+    });
+  }
+}
+
+// Read input into memory for now, as it's difficult to stream multipart
+// requests and this is what the backend uses.
+//
+// The interface takes streams to allow streaming later.
+async function streamToBlob(stream: ReadableStream<Uint8Array>): Promise<Blob> {
+  const chunks: Uint8Array[] = [];
+  const reader = stream.getReader();
+  while (true) {
+    const { done, value } = await reader.read();
+
+    if (value) {
+      chunks.push(value);
+    }
+
+    if (done) {
+      break;
+    }
+  }
+
+  return new Blob(chunks);
```

Comment:
Worth measuring but in theory it would be more efficient to avoid bouncing back and forth between C++ and JS:

```suggestion
  return new Response(stream).blob();
```

Reply:
Made this change, also much more obviously correct which is a plus for me!